File: apps/backend-java/src/main/java/com/housingplatform/features/profile/dto/ProfileDto.java
package com.housingplatform.features.profile.dto;

import com.housingplatform.persistence.enums.UserRole;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import java.time.OffsetDateTime;
import java.util.UUID;

public record ProfileDto(
    UUID id,
    UserRole role,
    String fullName,
    String phone,
    String avatarUrl,
    String preferredLanguage,
    boolean marketingConsent,
    OffsetDateTime createdAt,
    OffsetDateTime updatedAt) {}

File: apps/backend-node/src/features/profile/dto/profile.dto.ts
import type { UserRole } from '../../../shared/auth/auth-user.model';

export interface ProfileDto {
  id: string;
  role: UserRole;
  fullName: string;
  phone: string | null;
  avatarUrl: string | null;
  preferredLanguage: string;
  marketingConsent: boolean;
  createdAt: string;
  updatedAt: string;
}

export interface UpdateProfileDto {
  fullName: string;
  phone?: string | null;
  preferredLanguage: string;
  marketingConsent: boolean;
  avatarUrl?: string | null;
}
